azure-pipelines/dotnet-vscode-csharp-insertion.yml

```diff
@@ -0,0 +1,101 @@
+  - name: RoslynEndSHA
+    value: $(resources.pipeline.officialBuildCI.sourceCommit)
+  - name: RoslynBuildNumber
+    value: $(resources.pipeline.officialBuildCI.runName)
+  - name: RoslynBuildId
+    value: $(resources.pipeline.officialBuildCI.runID)
+  - template: dotnet-variables.yml
+
+extends:
+  template: v1/1ES.Unofficial.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: netcore1espool-internal
+      image: 1es-ubuntu-2204
+      os: linux
+    stages:
+      - stage: BumpRoslyn
+        displayName: Bump Roslyn Version
+        jobs:
+          - job: ProcessBump
+            displayName: Process Roslyn Bump
+            pool:
+              name: netcore1espool-internal
+              image: 1es-ubuntu-2204
+              os: linux
+            steps:
+              - checkout: self
+                persistCredentials: true
+
+              - task: UseDotNet@2
+                displayName: Install .NET SDK
+                inputs:
+                  version: $(defaultDotnetVersion)
+
+              - template: install-node.yml
+
+              - task: DownloadPipelineArtifact@2
+                displayName: Download Asset Manifests
+                inputs:
+                  source: specific
+                  project: internal
+                  pipeline: dotnet-roslyn-official
+                  runVersion: specific
+                  runId: $(RoslynBuildId)
+                  artifact: AssetManifests
+                  path: $(Pipeline.Workspace)/AssetManifests
+
+              - pwsh: |
+                  npm ci
+                  npm install
+                  npm install -g gulp
+                  gulp installDependencies
+                displayName: 'Install npm dependencies and gulp'
+              - checkout: RoslynMirror
+                displayName: 'Checkout Roslyn repository'
+                path: roslyn
+
+              - script: |
+                  echo "Installing roslyn-tools..."
+                  dotnet tool install -g Microsoft.RoslynTools --prerelease --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json
+                  echo "Verifying roslyn-tools installation..."
+                  dotnet tool list -g
+                  roslyn-tools --version
+                displayName: 'Install roslyn-tools CLI'
+
+              - task: Npm@1
+                displayName: Run Roslyn insertion
+                env:
+                  GITHUB_TOKEN: $(System.AccessToken)
+                  GitHubPAT: $(System.AccessToken)
+                inputs:
+                  command: custom
+                  customCommand: 'run gulp -- insertion:roslyn --assetManifestPath=$(Pipeline.Workspace)/AssetManifests --roslynRepoPath=$(Pipeline.Workspace)/roslyn --roslynEndSHA=$(RoslynEndSHA) --roslynBuildNumber=$(RoslynBuildNumber) --roslynBuildId=$(RoslynBuildId) --targetBranch=${{ parameters.targetBranch }} --githubPAT=$(System.AccessToken) --dryRun=false'
```

Comment:
How does this PAT flow in here? Do you set it on the pipeline or is this something that is coming in from a variable group?

`--githubPAT=$(System.AccessToken)`

Reply:
Since you are running in dnceng you should be able to use the same PAT as the localization pipeline - https://github.com/dotnet/vscode-csharp/blob/29cb53a66cbccfd79655e2825b4c2b82c668ef19/azure-pipelines/loc.yml#L95